### PR TITLE
chore: Add kms, crypto, couch db to server start-up

### LINKED ISSUE #19: Sign Anchor Credential Setup
Batch writer should sign anchor credential before creating an offer for witnesses. In order to do this we have to setup kms, crypto, couch db during server start-up.

### PR DESCRIPTION
**Pull request by sandrask**
- Refactor reading of orb server arguments (aligned with other Trustbloc projects)
- Create kms and crypto providers during orb services start-up
- Add unit-tests for cmd/orb-server

Closes #19

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: cmd/orb-server/main.go [hunk 1]
```diff
@@ -7,173 +7,25 @@ SPDX-License-Identifier: Apache-2.0
...
-
-var config = viper.New()
-
-const (
-	defaultDIDDocNamespace = "did:orb"
-	basePath               = "/sidetree/0.0.1"
-	arrayDelimiter         = ","
-	txnBuffer              = 100
-)
+var logger = log.New("orb-rest")
```
Comment (troyronda) [inline cmd/orb-server/main.go:16]: We should rename rest to server throughout.

File: cmd/orb-server/startcmd/params.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+	tokenFlagName  = "api-token"
+	tokenEnvKey    = "ORB_API_TOKEN" //nolint: gosec
+	tokenFlagUsage = "Check for bearer token in the authorization header (optional). " +
+		commonEnvVarUsageText + tokenEnvKey
+
+	databaseTypeMemOption     = "mem"
+	databaseTypeCouchDBOption = "couchdb"
+	databaseTypeMYSQLDBOption = "mysql"
+)
```
Comment (troyronda) [inline cmd/orb-server/startcmd/params.go:91]: Todo: Postgres

**Review by troyronda — APPROVED** [review]

File: cmd/orb-server/main.go [hunk 3]
```diff
@@ -7,173 +7,25 @@ SPDX-License-Identifier: Apache-2.0
...
-	config.SetEnvPrefix("ORB")
-	config.AutomaticEnv()
-	config.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
-
-	didDocNamespace := defaultDIDDocNamespace
-	if config.GetString("did.namespace") != "" {
-		didDocNamespace = config.GetString("did.namespace")
+func main() {
+	rootCmd := &cobra.Command{
+		Use: "vc-rest",
```
Comment (troyronda) [inline cmd/orb-server/main.go:20]: Is it supposed to be orb-server?

**Review by troyronda — APPROVED** [review]

File: cmd/orb-server/main.go (hunk 1 above)
Comment (sandrask) [reply cmd/orb-server/main.go:16]: sure, next PR

File: cmd/orb-server/startcmd/params.go (hunk 2 above)
Comment (sandrask) [reply cmd/orb-server/startcmd/params.go:91]: Will create TODO

File: cmd/orb-server/main.go (hunk 3 above)
Comment (sandrask) [reply cmd/orb-server/main.go:20]: yes, I will rename it to orb-server in my next PR

**Pull request merged**